pkg/controller/jobs/deployment/deployment_controller.go

```diff
@@ -38,18 +39,26 @@ const (
 
 func init() {
 	utilruntime.Must(jobframework.RegisterIntegration(FrameworkName, jobframework.IntegrationCallbacks{
-		SetupIndexes:   SetupIndexes,
-		NewReconciler:  jobframework.NewNoopReconcilerFactory(gvk),
-		GVK:            gvk,
-		SetupWebhook:   SetupWebhook,
-		JobType:        &appsv1.Deployment{},
-		AddToScheme:    appsv1.AddToScheme,
-		DependencyList: []string{"pod"},
+		SetupIndexes:           SetupIndexes,
+		NewReconciler:          jobframework.NewNoopReconcilerFactory(gvk),
+		GVK:                    gvk,
+		SetupWebhook:           SetupWebhook,
+		JobType:                &appsv1.Deployment{},
+		AddToScheme:            appsv1.AddToScheme,
+		DependencyList:         []string{"pod"},
+		IsManagingObjectsOwner: isDeployment,
```

Comment:
This looks ok-ish, but I have no clue why we need to change it, please clarify.

Reply:
We need this for case `Pod -> Deployment -> AppWrapper`.

Reply:
I don't understand why - can you clarify.

Reply:
Yes, this function is using to compare ownerReference with Deployment.

https://github.com/kubernetes-sigs/kueue/blob/45c67c81a3dd21a86c84dc7384556658bdd72caf/pkg/controller/jobframework/integrationmanager.go#L190-L196

https://github.com/kubernetes-sigs/kueue/blob/45c67c81a3dd21a86c84dc7384556658bdd72caf/pkg/controller/jobframework/integrationmanager.go#L339-L346

If this is true, we can return the JobType, and we can get empty Object for this Job. If false thats means that this this integration is not managed by Kueue.

https://github.com/kubernetes-sigs/kueue/blob/45c67c81a3dd21a86c84dc7384556658bdd72caf/pkg/controller/jobframework/reconciler.go#L637-L638

Reply:
ack, thanks, I see it is implemented analogously for other types, such as Job and JobSet. Feel free to resolve.

Reply:
Opened https://github.com/kubernetes-sigs/kueue/issues/4940  to explore it more and cleanup potentially.